### DISCUSSION TITLE
Beta suggestion: Tariff inject

### DISCUSSION THREAD

**Discussion by himynamesned** [netzero-labs/netzero#56] [Ideas]
Octopus Energy Free Electricity sessions. Possible to add a beta for polling the following JSON? They have seemed to keep the API up to date, despite no official API for free electricity sessions from OE directly. The JSON was updated by time I had the email from OE...

This could be a check box in the app to inject free sessions, when available, if NetZero is already managing the tariff on the powerwall. 

https://www.whizzy.org/2024-09-14-free-electricity-sessions/

**Comment by zigam** [comment]
Hi,

Thanks for the suggestion, we'll consider this. You need to opt in to the free electricity session however, right?

**Reply by himynamesned** [reply]
Correct. It's a one-off enroll. Not per session (as it used to be).

![image](https://github.com/user-attachments/assets/f0211144-2a34-46df-b0ec-ee6b5b4862d9)

Thanks!

**Reply by himynamesned** [reply]
Bump on this please. Kind of related to the other tariff manipulation discussions

**Reply by zigam** [reply]
Similarly to the Netzero IOG integration, this integration would automatically charge up the Powerwall during a free electricity session, correct? Also, our understanding is that free electricity sessions cannot be used with Intelligent tariffs, those have different rules.

We'll contact the author of the JSON API to see if use is allowed.

**Reply by himynamesned** [reply]
Yes that's correct. You could get "smarter" and have it look ahead to ensure it max exports prior to a free event, to maximise savings etc- although TBC kind of does this already, automatically. (see my other thread where i've manipulated max export and imports to make use of when export £ is greater than import £, and cycled charge/discharge cycles via tariff £ manipulation)

Yes but fine with the likes of Go tariff.

**Reply by himynamesned** [reply]
Thanks for including this into the beta @zigam - look forward to testing.
